gobblin-compaction/src/main/java/gobblin/compaction/conditions/RecompactionConditionBasedOnPercent.java

```diff
@@ -0,0 +1,32 @@
+package gobblin.compaction.conditions;
+
+
+import gobblin.compaction.dataset.DatasetMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+/**
+ * An implementation {@link RecompactionCondition} which examines the late record percentage.
+ * If the percent exceeds the limit, a recompaction is triggered.
+ */
+public class RecompactionConditionBasedOnPercent implements RecompactionCondition {
+
+  private static final Logger logger = LoggerFactory.getLogger(RecompactionConditionBasedOnTime.class);
+  private double percent = 0;
```

Comment:
Based on the implementation, it is better to call it `ratio`, rather than `percent`.